### PR TITLE
Better trimming support + trimming fix

### PR DESCRIPTION
**Pull request by mmitche**
- Allow for trimming to be set in SourceBuild.props. This enables use in repo SB legs. Includes a default set which is all netcoreapps + netstandards.
- Disable trimming for top level projects in Arcade, so that we don't accidentally skip things like publishing.

### To double check:

* [ ] The right tests are in and and the right validation has happened.  Guidance:  https://github.com/dotnet/arcade/tree/main/Documentation/Validation

### CODE REVIEW

**Comment by mmitche** [comment]
One question I had was why Tools.proj looks like this:

https://github.com/dotnet/arcade/blob/main/src/Microsoft.DotNet.Arcade.Sdk/tools/Tools.proj#L6-L13

I didn't disable trimming in this project yet.  I think it might mean that the NuGetRepack pack will not get restored. Anyways, my  main question is why this project explicitly sets a bunch of TFM properties.

**Comment by NikolaMilosavljevic** [comment]
Just curious, where are these properties consumed? `NoTargetFrameworkFiltering` and `DotNetTargetFrameworkFilter`

**Comment by mmitche** [comment]
https://github.com/dotnet/arcade/blob/main/src/Microsoft.DotNet.Arcade.Sdk/tools/TargetFrameworkFilters.BeforeCommonTargets.targets#L4-L18

**Comment by NikolaMilosavljevic** [comment]
@MichaelSimons

**Review by NikolaMilosavljevic — APPROVED** [review]

**Comment by mmitche** [comment]
@NikolaMilosavljevic Slight tweak here to the default TFM string (need re-approval). This did work in source build with it enabled to sdk +installer.

File: src/Microsoft.DotNet.Arcade.Sdk/tools/SourceBuild/SourceBuildArcade.targets [hunk 1]
```diff
@@ -28,6 +28,11 @@
     <DetectSourceBuildIntermediateNupkgLicense Condition="'$(DetectSourceBuildIntermediateNupkgLicense)' == ''">true</DetectSourceBuildIntermediateNupkgLicense>
 
     <EnableDefaultSourceBuildIntermediateItems Condition="'$(EnableDefaultSourceBuildIntermediateItems)' == ''">true</EnableDefaultSourceBuildIntermediateItems>
+
+    <!-- This is fed into the inner source build as an environment variable (DotNetTargetFrameworkFilter) to enable filtering.
+         The default target framework filter includes all recent netcoreapps in SBRP, as well as 8.0 (latest) and 7.0 (working to get rid of it -->
+    <_DefaultNetFrameworkFilter>netstandard2.0%3bnetstandard2.1%3bnetcoreapp2.1%3bnet5.0%3bnet6.0%3bnet7.0%3bnet8.0</_DefaultNetFrameworkFilter>
```
Comment (NikolaMilosavljevic) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/SourceBuild/SourceBuildArcade.targets:34]: netcoreapp3.1 is missing - presumably intentional?

Comment (mmitche) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/SourceBuild/SourceBuildArcade.targets:34]: Ahh, shouldn't be. Good catch.

**Pull request merged**